src/libaktualizr/uptane/uptane_network_test.cc

```diff
@@ -139,6 +141,85 @@ TEST(UptaneNetwork, DownloadFailure) {
   EXPECT_TRUE(result.first);
 }
 
+/*
+ * Output a log when connectivity is restored.
+ */
+class HttpUnstable : public HttpFake {
+ public:
+  HttpUnstable(const boost::filesystem::path &test_dir_in) : HttpFake(test_dir_in, "hasupdates") {}
+  HttpResponse get(const std::string &url, int64_t maxsize) override {
+    if (!connectSwitch) {
+      return HttpResponse({}, 503, CURLE_OK, "");
+    } else {
+      return HttpFake::get(url, maxsize);
+    }
+  }
+
+  HttpResponse put(const std::string &url, const Json::Value &data) override {
+    if (!connectSwitch) {
+      (void)data;
+      return HttpResponse(url, 504, CURLE_OK, "");
+    } else {
+      last_manifest = data;
+      return HttpResponse(url, 200, CURLE_OK, "");
+    }
+  }
+
+  HttpResponse post(const std::string &url, const Json::Value &data) override {
+    if (!connectSwitch) {
+      (void)data;
+      return HttpResponse(url, 503, CURLE_OK, "");
+    } else {
+      if (url.find("/devices") != std::string::npos || url.find("/director/ecus") != std::string::npos || url.empty()) {
```

Comment:
Is there a reason why we can't call `HttpFake::post()` here? Copying this logic here, would complicate maintenance.

Reply:
I'm not sure, but if will fail if I use the original. I'll have a test.

Reply:
```
file served: "/tmp/aktualizr-4d8c-c7cb-fcae-9383/619b-47da-dir/repo/targets_hasupdates.json"
4429
[2019-12-09 12:59:48.635639] [0x000000000404fcc0] [info]    2 new updates found in Uptane metadata.
4430
[2019-12-09 12:59:48.659888] [0x000000000404fcc0] [info]    got UpdateCheckComplete event
4431
[2019-12-09 12:59:50.232787] [0x000000000404fcc0] [error]   Failed to update director metadata: 
4432
[2019-12-09 12:59:50.236768] [0x000000000404fcc0] [info]    got UpdateCheckComplete event
4433
../src/libaktualizr/uptane/uptane_network_test.cc:217: Failure
4434
Expected: (std::string::npos) != (testing::internal::GetCapturedStdout().find("Connectivity is restored.")), actual: 18446744073709551615 vs 1844674407370955161
```

Reply:
Well, you still need to handle the failure case. In the last version you've pushed, `put` and `post` are the same as in HttpFake.

If you follow the same pattern that what you did for `get()`, it should work.

e.g:

```
HttpResponse post(const std::string &url, const Json::Value &data) override {
 if (!connectSwitch) {
   (void)data;
   return HttpResponse(url, 503, CURLE_OK, "");
 } else {
   return HttpFake::post(url, data);
  }
}
```

Reply:
Oh, I see, thank you!